vm/devices/user_driver/src/buffer_manager.rs

```diff
@@ -0,0 +1,259 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! Implementation of a buffer manager for managing memory blocks.
```

Comment:
I imagine you already looked, but I would feel remiss if I did not ask: is there no existing crate that does this already? This seems like a generic primitive.